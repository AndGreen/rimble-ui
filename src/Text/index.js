import React from 'react'
import styled from 'styled-components'

import { injectGlobal } from 'styled-components';

injectGlobal`
  :root {
    font-family: "Inter UI";
    font-size: 16px;
  }
`

import {
  fontSize,
  space,
  color,
  responsiveStyle
} from 'styled-system'



const Text = styled.div`
  ${fontSize} ${space} ${color}
`

Text.span = Text.withComponent('span')
Text.p = Text.withComponent('p')
Text.s = Text.withComponent('s')

export default Text
