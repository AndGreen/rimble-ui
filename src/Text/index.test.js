import Text from './'

describe('Text component sanity', () => {
  it('has name', () => {
    expect(Text.displayName).toBe('Text')
  })

  it('matches default snapshot', () => {
    const tree = renderer.create(<Text />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
