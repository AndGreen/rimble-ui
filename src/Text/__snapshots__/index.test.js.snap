// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component matches default snapshot 1`] = `
.c0 {
  color: copyColor;
  font-size: 16px;
  font-size: 16px;
  font-family: sansSerif;
  font-weight: 2;
  text-align: left;
  line-height: copy;
}

.c0 {
  box-sizing: border-box;
}

<div
  className="c0"
  color="copyColor"
  fontFamily="sansSerif"
  fontSize={2}
  fontWeight={2}
/>
`;

exports[`Text component sanity matches default snapshot 1`] = `
.c0 {
  color: copyColor;
  font-size: 16px;
  font-size: 16px;
  font-family: sansSerif;
  font-weight: 2;
  text-align: left;
  line-height: copy;
}

.c0 {
  box-sizing: border-box;
}

<div
  className="c0"
  color="copyColor"
  fontFamily="sansSerif"
  fontSize={2}
  fontWeight={2}
/>
`;
